libcst/_types.py

```diff
@@ -4,11 +4,13 @@
 # LICENSE file in the root directory of this source tree.
 
 
-from typing import TYPE_CHECKING, TypeVar
+from pathlib import Path
+from typing import TYPE_CHECKING, TypeVar, Union
 
 if TYPE_CHECKING:
     from libcst._nodes.base import CSTNode  # noqa: F401
 
 
 CSTNodeT = TypeVar("CSTNodeT", bound="CSTNode")
 CSTNodeT_co = TypeVar("CSTNodeT_co", bound="CSTNode", covariant=True)
+StrPath = Union[str, Path]
```

Comment:
Would `os.PathLike` suffice?

Reply:
Turns out that `os.PathLike` behaves differently (and mutually incompatible from a typing perspective) between 3.8 and 3.9. So I went ahead and reverted back to pathlib, though using `PurePath` in hopes of covering more potential usage. We can revisit in the future if we drop support for older runtimes, or if there's a better alternative available.

Reply:
TIL